appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java

```diff
@@ -116,6 +125,38 @@ public void refresh() {
                 .sorted().toList();
         group.setItems(sortedGroups);
         applyFilters();
+
+        prepareExportData(dependencies);
+    }
+
+    private void prepareExportData(List<Dependency> dependencyList) {
+        try (
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                CSVPrinter printer = new CSVPrinter(new OutputStreamWriter(outputStream), CSVFormat.DEFAULT)
+        ) {
+            // header
+            printer.printRecord("Dependency", "Ecosystem", "Dependency group", "Version", "Is development?", "# of vulnerabilities", "Highest severity", "Highest CVSS score");
+            // content
+            for (Dependency dependency : dependencyList) {
+                printer.printRecord(
+                        dependency.getName(),
+                        dependency.getEcosystem(),
+                        dependency.getGroup(),
+                        dependency.getVersion(),
+                        dependency.isDevDependency(),
+                        dependency.getNumOfVulnerabilities(),
+                        dependency.getSeverityLevel(),
+                        dependency.getRiskScore()
+                );
+            }
+
+            String fileName = "dependencies.csv";
+            StreamResource streamResource = new StreamResource(fileName, () -> new ByteArrayInputStream(outputStream.toByteArray()));
+            updateExportData(streamResource);
+        } catch (IOException e) {
+            // TODO handle exception properly
+            e.printStackTrace();
```

Comment:
I am not sure what to do here.

Reply:
You could throw an [AppSecException](https://github.com/vaadin/appsec-kit/blob/main/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecException.java#L20) with some message.

Reply:
I see. Then, what about showing an error log in dev tools' log tab and also logging an error or warn in the logs?

Reply:
But maybe an exception would be more suitable for these kind of errors. We could create an `AppSecException` in starter module too or make the other one public. What do you think? (cc @heruan )

Reply:
If there an error during the export, the application shouldn't throw. Let's catch the exception as it is and push a notification to the UI with the error message, to make it clear for the user that something went wrong.